src/r_segs.c

```diff
@@ -525,9 +525,10 @@ void R_StoreWallRange(const int start, const int stop)
 
   if (ds_p == drawsegs+maxdrawsegs)   // killough 1/98 -- fix 2s line HOM
     {
+      ptrdiff_t pos = ds_p - drawsegs; // jff 8/9/98 fix from ZDOOM1.14a
       unsigned newmax = maxdrawsegs ? maxdrawsegs*2 : 128; // killough
       drawsegs = Z_Realloc(drawsegs,newmax*sizeof(*drawsegs),PU_STATIC,0);
-      ds_p = drawsegs+maxdrawsegs;
+      ds_p = drawsegs + pos;          // jff 8/9/98 fix from ZDOOM1.14a
```

Comment:
But at this point `pos == maxdrawsegs` or not?

Reply:
You're right, I think we don't need this change. Fixed in 32d1469